### PR TITLE
remove packedDependencies and custom package script

### PR DESCRIPTION
**Pull request by jackkav**
uses electron-builder conventions to avoid maintaining the packedDependencies list and release.ts
the subtle conventional change is that insomnia-app/package.json devDependencies are all bundled by vite and dependencies will be installed inside the final package as before.

### CODE REVIEW

File: packages/insomnia-app/.electron-builder.config.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/**
+ * @type {import('electron-builder').Configuration}
```
Comment (dimitropoulos) [inline packages/insomnia-app/.electron-builder.config.js]: good call!  nice to have validation!

File: packages/insomnia-app/.electron-builder.config.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * @see https://www.electron.build/configuration/configuration
+ */
+// NOTE: USE_HARD_LINKS
+// https://github.com/electron-userland/electron-builder/issues/4594#issuecomment-574653870
+const config = {
+    appId: "com.insomnia.app",
```
Comment (dimitropoulos) [inline packages/insomnia-app/.electron-builder.config.js]: this was `__APP_ID__` before.  is there any significance to it being `com.insomnia.app` now?

File: packages/insomnia-app/config/electronbuilder.json [hunk 3]
```diff
@@ -1,89 +0,0 @@
...
-  "squirrelWindows": {
-    "artifactName": "__BINARY_PREFIX__-${version}.${ext}",
-    "iconUrl": "__ICON_URL__"
-  },
-  "portable": {
-    "artifactName": "__BINARY_PREFIX__-${version}-portable.${ext}"
-  },
-  "linux": {
-    "artifactName": "__BINARY_PREFIX__-${version}.${ext}",
-    "executableName": "__EXECUTABLE_NAME__",
```
Comment (dimitropoulos) [inline packages/insomnia-app/config/electronbuilder.json:78]: should we also then delete this from `config.json`?

**Review by dimitropoulos — COMMENTED** [review]
What a time to be alive! This is exciting!

I spent a good deal of time just now looking at this and trying to find a hole in it, but I see none.  Now, I'm not an expert on electron-builder, but this sure does seem to be a more "normal" way to use it.

I added a few (very very minor) tweaks (see commit message and description for more on what or why).

_____

The only thing that is of some concern is that I noticed that the before-and-after of the output is different, somewhat significantly so.  Can someone else take a look at this?

Before this PR I get something like this:
```
insomnia/packages/insomnia-app/dist$ find . -not -path "*node_modules*" -not -path "*locales*"
.
./Insomnia.Core-2022.3.0-beta.3.AppImage
./Insomnia.Core-2022.3.0-beta.3.snap
./Insomnia.Core-2022.3.0-beta.3.tar.gz
./.icon-set
./.icon-set/icon_48x48.png
./.icon-set/icon_256.png
./.icon-set/icon_128x128.png
./.icon-set/icon_512.png
./.icon-set/icon_32x32.png
./.icon-set/icon_64x64.png
./.icon-set/icon_16x16.png
./builder-effective-config.yaml
./linux-unpacked
./linux-unpacked/LICENSE.electron.txt
./linux-unpacked/chrome-sandbox
./linux-unpacked/v8_context_snapshot.bin
./linux-unpacked/libvulkan.so.1
./linux-unpacked/icudtl.dat
./linux-unpacked/snapshot_blob.bin
./linux-unpacked/chrome_200_percent.pak
./linux-unpacked/libGLESv2.so
./linux-unpacked/resources
./linux-unpacked/resources/bin
./linux-unpacked/resources/bin/yarn-standalone.js
./linux-unpacked/resources/opensource-licenses.txt
./linux-unpacked/resources/app.asar
./linux-unpacked/swiftshader
./linux-unpacked/swiftshader/libGLESv2.so
./linux-unpacked/swiftshader/libEGL.so
./linux-unpacked/libffmpeg.so
./linux-unpacked/resources.pak
./linux-unpacked/chrome_crashpad_handler
./linux-unpacked/libvk_swiftshader.so
./linux-unpacked/insomnia
./linux-unpacked/chrome_100_percent.pak
./linux-unpacked/libEGL.so
./linux-unpacked/vk_swiftshader_icd.json
./linux-unpacked/LICENSES.chromium.html

```

But after this PR I get something like this:
```
insomnia/packages/insomnia-app/dist$ find . -not -path "*node_modules*" -not -path "*locales*"
.
./builder-debug.yml
./builder-effective-config.yaml
./linux-unpacked
./linux-unpacked/LICENSE.electron.txt
./linux-unpacked/chrome-sandbox
./linux-unpacked/v8_context_snapshot.bin
./linux-unpacked/libvulkan.so.1
./linux-unpacked/icudtl.dat
./linux-unpacked/snapshot_blob.bin
./linux-unpacked/chrome_200_percent.pak
./linux-unpacked/libGLESv2.so
./linux-unpacked/resources
./linux-unpacked/resources/bin
./linux-unpacked/resources/bin/yarn-standalone.js
./linux-unpacked/resources/app
./linux-unpacked/resources/app/app
./linux-unpacked/resources/app/app/chart.383d9cac.svg
./linux-unpacked/resources/app/app/install-spinner.gif
./linux-unpacked/resources/app/app/index.c47a3bbf.js
./linux-unpacked/resources/app/app/fontawesome-webfont.7dacf83f.woff2
./linux-unpacked/resources/app/app/opensource-licenses.txt
./linux-unpacked/resources/app/app/static
./linux-unpacked/resources/app/app/static/insomnia-core-logo_16x.png
./linux-unpacked/resources/app/app/static/entitlements.mac.inherit.plist
./linux-unpacked/resources/app/app/icon.icns
./linux-unpacked/resources/app/app/main.min.js.map
./linux-unpacked/resources/app/app/icon.ico
./linux-unpacked/resources/app/app/insomnia-logo.a19543fc.svg
./linux-unpacked/resources/app/app/preload.js
./linux-unpacked/resources/app/app/background.png
./linux-unpacked/resources/app/app/index.html
./linux-unpacked/resources/app/app/icn-eye.070cdb80.svg
./linux-unpacked/resources/app/app/main.min.js
./linux-unpacked/resources/app/app/index.c9a30e89.css
./linux-unpacked/resources/app/app/yarn-standalone.js
./linux-unpacked/resources/app/package.json
./linux-unpacked/resources/opensource-licenses.txt
./linux-unpacked/swiftshader
./linux-unpacked/swiftshader/libGLESv2.so
./linux-unpacked/swiftshader/libEGL.so
./linux-unpacked/libffmpeg.so
./linux-unpacked/resources.pak
./linux-unpacked/chrome_crashpad_handler
./linux-unpacked/libvk_swiftshader.so
./linux-unpacked/insomnia
./linux-unpacked/chrome_100_percent.pak
./linux-unpacked/libEGL.so
./linux-unpacked/vk_swiftshader_icd.json
./linux-unpacked/LICENSES.chromium.html
```

Of most concern for me is that the `.AppImage` files are gone now, which was something I used a decent amount in the past.

**Comment by jackkav** [comment]
@dimitropoulos thanks for the thorough review.
part of the diff is explained by `-config.asar=false` which helped me determine the appropriate electron config for files and directories. if this is removed we should see an asar binary instead of the app folder, theres still some more to figure out with icons i suspect. feel free to continue if it interests you.

File: packages/insomnia-app/config/electronbuilder.json (hunk 3 above)
Comment (jackkav) [reply packages/insomnia-app/config/electronbuilder.json:78]: I'm not sure if config.json what the distinction is between it and constants.ts, perhaps just moving anything used into constants.ts would be a more convenient grouping.

File: packages/insomnia-app/.electron-builder.config.js (hunk 2 above)
Comment (jackkav) [reply packages/insomnia-app/.electron-builder.config.js]: thats just what package.ts replaced it with from config.json AFAIK
```js
const rawConfig = JSON.stringify(electronBuilderConfig, null, 2)
    .replace(/__APP_ID__/g, appId)
    .replace(/__BINARY_PREFIX__/g, binaryPrefix)
    .replace(/__EXECUTABLE_NAME__/g, executableName)
    .replace(/__GITHUB_OWNER__/g, githubOrg)
    .replace(/__GITHUB_REPO__/g, githubRepo)
    .replace(/__ICON_URL__/g, icon)
    .replace(/__SYNOPSIS__/g, synopsis);
```
the original package.ts may have some behaviors that I didn't capture in the electron-builder.config.js
like mac universal

Comment (dimitropoulos) [reply packages/insomnia-app/.electron-builder.config.js]: riggghhhhttt righhhhtttt forgot when I wrote this and un-forgot later but forgot to delete my comment -> thanks for confirming!

File: packages/insomnia-app/electron-builder.config.js [hunk 4]
```diff
@@ -87,8 +88,15 @@ const config = {
...
     category: 'Development',
-    target: ['AppImage', 'deb', 'tar.gz', 'rpm', 'snap'],
+    target: ['AppImage', 'deb', 'tar.gz', 'snap'],
   },
 };
 
+const { env: { BUILD_TARGETS }, platform } = process
+const targets = BUILD_TARGETS?.split(',')
+if (platform && targets) {
+  console.log('overriding build targets to: ', targets)
```
Comment (dimitropoulos) [inline packages/insomnia-app/electron-builder.config.js]: awesome, thanks!

**Review by johnwchadwick — APPROVED** [review]
LGTM

**Review by gatzjames — APPROVED** [review]
🚀

**Review by marckong — APPROVED** [review]

**Pull request merged**